src/util/filedownload.ts

```diff
@@ -0,0 +1,24 @@
+const isSafari = /^((?!chrome|android).)*safari/i.test(navigator.userAgent);
+
+export const fileDownload = (
+  element: HTMLElement,
+  href: string,
+  filename: string
+): void => {
+  const a = document.createElement("a");
+  a.target = "_blank";
+  a.href = href;
+  a.download = filename;
+
+  element.shadowRoot!.appendChild(a);
+
+  setTimeout(
+    () => {
+      a.click();
+    },
+    // Directy calling a.click() is not working in iOS, adding a delay works around that issue.
+    // https://github.com/home-assistant/frontend/issues/9374
+    isSafari ? 500 : 0
```

Comment:
Are you sure other browsers still work now? I think a lot of browsers only allow the download when it was initiated by a user action, and with the timeout in between the browser will probably no longer link it to the user action.

Reply:
Tested chrome and Firefox, which browsers don't like it?

Reply:
Those ^, I think the popup blocker would block them...

Reply:
K... I'll move it _move it_

Reply:
Let's just only use the timeout on safari to be save?

Reply:
yeah, that's what I ment